src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -95,6 +97,36 @@ private void OnStandbyOptionsChange()
             }
         }
 
+        private HttpClient CreateMetricsPublisherHttpClient()
+        {
+            var clientHandler = new HttpClientHandler();
+            clientHandler.ServerCertificateCustomValidationCallback = ValidateRemoteCertificate;
+            return new HttpClient(clientHandler);
+        }
+
+        private bool ValidateRemoteCertificate(HttpRequestMessage httpRequestMessage, X509Certificate2 certificate, X509Chain certificateChain, SslPolicyErrors sslPolicyErrors)
+        {
+            if (sslPolicyErrors == SslPolicyErrors.None)
+            {
+                return true;
+            }
+
+            bool validateCertificateResult = (sslPolicyErrors == SslPolicyErrors.RemoteCertificateNameMismatch) && certificateChain.Build(certificate);
```

Comment:
What about the SslPolicyErrors.RemoteCertificateNotAvailable error case? Can you change this code to do an if/else like the Kubernetes client, so the error cases are clearer?

Reply:
We want to fail authentication incase of remotecertificatenotavailable by design